src/Microsoft.AspNet.Authentication.Cookies/Events/CookieAuthenticationEvents.cs

```diff
@@ -7,108 +7,77 @@
 namespace Microsoft.AspNet.Authentication.Cookies
 {
     /// <summary>
-    /// This default implementation of the ICookieAuthenticationNotifications may be used if the 
+    /// This default implementation of the ICookieAuthenticationEvents may be used if the 
     /// application only needs to override a few of the interface methods. This may be used as a base class
     /// or may be instantiated directly.
     /// </summary>
-    public class CookieAuthenticationNotifications : ICookieAuthenticationNotifications
+    public class CookieAuthenticationEvents : ICookieAuthenticationEvents
     {
         /// <summary>
-        /// Create a new instance of the default notifications.
-        /// </summary>
-        public CookieAuthenticationNotifications()
-        {
-            OnValidatePrincipal = context => Task.FromResult(0);
-            OnResponseSignIn = context => { };
-            OnResponseSignedIn = context => { };
-            OnResponseSignOut = context => { };
-            OnApplyRedirect = context => context.Response.Redirect(context.RedirectUri);
-            OnException = context => { };
-        }
-
-        /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Func<CookieValidatePrincipalContext, Task> OnValidatePrincipal { get; set; }
+        public Func<CookieValidatePrincipalContext, Task> OnValidatePrincipal { get; set; } = context => Task.FromResult(0);
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieResponseSignInContext> OnResponseSignIn { get; set; }
+        public Action<CookieResponseSignInContext> OnResponseSignIn { get; set; } = context => { };
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieResponseSignedInContext> OnResponseSignedIn { get; set; }
+        public Action<CookieResponseSignedInContext> OnResponseSignedIn { get; set; } = context => { };
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieResponseSignOutContext> OnResponseSignOut { get; set; }
+        public Action<CookieResponseSignOutContext> OnResponseSignOut { get; set; } = context => { };
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieApplyRedirectContext> OnApplyRedirect { get; set; }
+        public Action<CookieApplyRedirectContext> OnApplyRedirect { get; set; } = context => context.Response.Redirect(context.RedirectUri);
```

Comment:
Yep, since OIDC's `RedirectToIdentityProvider` or `SecurityTokenValidated` events have their default behavior directly in the handler (I'm sure they are not the only ones) :smile: